subcommand/consul-sidecar/command.go

```diff
@@ -156,6 +156,7 @@ func (c *Command) Run(args []string) int {
 		go func() {
 			if err = server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
 				c.logger.Error(err.Error())
+				os.Exit(1)
```

Comment:
should this just be `return 1` instead?

Reply:
Actually, os.Exit(1) won't work in a goroutine. I need to somehow signal to Run to exit.

Reply:
See the command/inject-connect for how we do this. It uses a channel that the error gets sent to.